framework/test/mocha/unit/modules/chain/components/storage/entitites/transaction.js

```diff
@@ -0,0 +1,893 @@
+			const block = seeder.getLastBlock();
+			const transactions = [];
+			for (let i = 0; i < numSeedRecords; i++) {
+				transactions.push(
+					new transactionsFixtures.Transaction({
+						blockId: block.id,
+						type: transactionTypes.DAPP,
+					})
+				);
+			}
+			await storage.entities.Transaction.create(transactions);
+			const transactionIds = transactions.map(({ id }) => id);
+			const result = await storage.entities.Transaction.get(
+				{ id_in: transactionIds },
+				{ extended: true }
+			);
+
+			expect(result).to.not.empty;
+			expect(result).to.have.lengthOf(numSeedRecords);
+			expect(result.map(r => r.id)).to.be.eql(transactions.map(t => t.id));
+			expect(result.map(t => t.asset.dapp)).to.be.eql(
+				transactions.map(t => t.asset.dapp)
+			);
+		});
+
+		it('should populate asset field with "intransfer" json for type 6 transactions', async () => {
+			const block = seeder.getLastBlock();
+			const transactions = [];
+			for (let i = 0; i < numSeedRecords; i++) {
+				transactions.push(
+					new transactionsFixtures.Transaction({
+						blockId: block.id,
+						type: transactionTypes.IN_TRANSFER,
+					})
+				);
+			}
+			await storage.entities.Transaction.create(transactions);
+			const transactionIds = transactions.map(({ id }) => id);
+			const result = await storage.entities.Transaction.get(
+				{ id_in: transactionIds },
+				{ extended: true }
+			);
+
+			expect(result).to.not.empty;
+			expect(result).to.have.lengthOf(numSeedRecords);
+			expect(result.map(r => r.id)).to.be.eql(transactions.map(t => t.id));
+			expect(result.map(r => r.asset.inTransfer.transactionId)).to.be.eql(
+				transactions.map(t => t.asset.inTransfer.transactionId)
+			);
+			expect(result.map(r => r.asset.inTransfer.dappId)).to.be.eql(
+				transactions.map(t => t.asset.inTransfer.dappId)
+			);
+		});
+
+		it('should populate asset field with "outtransfer" json for type 7 transactions', async () => {
+			const block = seeder.getLastBlock();
+			const transactions = [];
+			for (let i = 0; i < numSeedRecords; i++) {
+				transactions.push(
+					new transactionsFixtures.Transaction({
+						blockId: block.id,
+						type: transactionTypes.OUT_TRANSFER,
+					})
+				);
+			}
+			await storage.entities.Transaction.create(transactions);
+			const transactionIds = transactions.map(({ id }) => id);
+			const result = await storage.entities.Transaction.get(
+				{ id_in: transactionIds },
+				{ extended: true }
+			);
+
+			expect(result).to.not.empty;
+			expect(result).to.have.lengthOf(numSeedRecords);
+			expect(result.map(r => r.id)).to.be.eql(transactions.map(t => t.id));
+			expect(result.map(r => r.asset.outTransfer.transactionId)).to.be.eql(
+				transactions.map(t => t.asset.outTransfer.transactionId)
+			);
+			expect(result.map(r => r.asset.outTransfer.dappId)).to.be.eql(
+				transactions.map(t => t.asset.outTransfer.dappId)
+			);
+		});
+	});
+
+	describe('get()', () => {
```

Comment:
`chain/storage` tests should only test what it implements and it doesn't implement getters.
I suggest moving the getters tests to `component/storage` test